src/hooks/useGetTokensBalance.ts

```diff
@@ -19,65 +25,92 @@ export const useGetTokensBalance = (
   const { accountId } = useWalletSelector()
   const { activePreview } = useHistoryStore((state) => state)
   const { data: exchangesList, isFetched } = useGetCoingeckoExchangesList()
+  const { getAccountBalance } = useAccountBalance()
+
+  const isTokenNative = (address: string): boolean => {
+    switch (address) {
+      case "native":
+        return true
+      default:
+        return false
+    }
+  }
+
+  const getNearTokenBalance = async (
+    address: string,
+    decimals: number
+  ): Promise<TokenBalance> => {
+    const tokenBalance: TokenBalance = {}
+    let balance: number | undefined = undefined
+
+    if (accountId && isTokenNative(address)) {
+      const { balance } = await getAccountBalance()
+      if (balance) {
+        const formattedAmountOut = formatUnits(BigInt(balance), decimals)
+        Object.assign(tokenBalance, { balance: parseFloat(formattedAmountOut) })
+      }
+    } else if (accountId && !isTokenNative(address)) {
+      const getBalance = await nep141Balance(accountId as string, address)
+      if (getBalance) {
+        balance = Number(formatUnits(BigInt(getBalance as string), decimals))
+        Object.assign(tokenBalance, { balance })
+      }
+    }
+
+    const getCoinIndex = (
```

Comment:
Feels like fetching USD value could be separated , to simplify everything.

Reply:
I guess this is because of misleading in function name. The function does to us USD balance and Rate, so we can keep is as it is and rename it as getUSDEquivalentAndRate or divide to two separate getUSDValue and getTokenRate.